packages/common/src/api/tan-query/useTrendingPlaylists.ts

```diff
@@ -0,0 +1,95 @@
+import { OptionalId } from '@audius/sdk'
+import { useInfiniteQuery, useQueryClient } from '@tanstack/react-query'
+import { useDispatch } from 'react-redux'
+
+import {
+  transformAndCleanList,
+  userCollectionMetadataFromSDK
+} from '~/adapters'
+import { useAudiusQueryContext } from '~/audius-query'
+import { PlaybackSource } from '~/models/Analytics'
+import {
+  trendingPlaylistsPageLineupActions,
+  trendingPlaylistsPageLineupSelectors
+} from '~/store/pages'
+
+import { QueryOptions } from './types'
+import { useCurrentUserId } from './useCurrentUserId'
+import { loadNextPage } from './utils/infiniteQueryLoadNextPage'
+import { primeCollectionData } from './utils/primeCollectionData'
+import { useLineupQuery } from './utils/useLineupQuery'
+
+const DEFAULT_PAGE_SIZE = 10
+
+type UseTrendingPlaylistsArgs = {
+  pageSize?: number
+}
+
+export const useTrendingPlaylists = (
+  { pageSize = DEFAULT_PAGE_SIZE }: UseTrendingPlaylistsArgs = {},
+  config?: QueryOptions
+) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const { data: currentUserId } = useCurrentUserId()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+
+  const queryData = useInfiniteQuery({
+    queryKey: ['trendingPlaylists', pageSize],
+    initialPageParam: 0,
+    getNextPageParam: (lastPage, allPages) => {
+      if (lastPage.length < pageSize) return undefined
+      return allPages.length * pageSize
+    },
+    queryFn: async ({ pageParam }) => {
+      const sdk = await audiusSdk()
+
+      const { data = [] } = await sdk.full.playlists.getTrendingPlaylists({
+        userId: OptionalId.parse(currentUserId),
+        limit: pageSize,
+        offset: pageParam,
+        time: 'week' as const
```

Comment:
should this be an arg?

Reply:
i was considering it! but it's not currently used that way. ill go ahead and make it an arg and default to week

Reply:
done